app/models/gene.rb

```diff
@@ -68,6 +68,11 @@ def species
   # CLASS INSTANCE METHODS
   ##
 
+  # find if a study has a given gene quickly for search gating
+  def self.study_has?(study_id, expr_matrix_ids, gene_name)
+    Gene.where(study_id: study_id, :study_file_id.in => expr_matrix_ids).any_of({name: gene_name},{searchable_name: gene_name}).exists?
+  end
+
```

Comment:
The post-rails 4 preferred syntax for searching where something is in an array is just `Gene.where(study_id: study_id, study_file_id: expr_matrix_ids) ` .   is that applicable for Mongoid here?

Reply:
Mongoid has different syntax for array queries.